data/mods/ssb/moves.ts

```diff
@@ -132,6 +132,47 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
 		type: "Ice",
 	},
 
+// Cake
+	kevin: {
+		accuracy: 100,
+		basePower: 111,
+		category: "Physical",
+		desc: "This move combines the user's current typing in its type effectiveness against the target.",
+		shortDesc: "Combines current types in its type effectiveness.",
+		name: "Kevin",
+		pp: 10,
+		flags: {protect: 1, mirror: 1},
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, 'Brave Bird', target);
+		},
+		onModifyType(move, pokemon, target) {
+			move.type = pokemon.types[0];
+		},
+		onTryImmunityPriority: -2,
+		onTryImmunity(target, pokemon) {
+			if (!target.runImmunity(pokemon.types[0])) return false;
```

Comment:
```suggestion
```

Reply:
dont actually commit this yet

Reply:
yeah this line is unnecessary as its already getting the effectiveness since the move changed type.